src/Security/Group.php

```diff
@@ -377,6 +377,53 @@ public function collateAncestorIDs()
     }
 
     /**
+     * Check if the group is a child of the given group or any parent groups
+     *
+     * @param string|int|Group $group Group instance, Group Code or ID
+     * @return bool Returns TRUE if the Group is a child of the given group, otherwise FALSE
+     */
+    public function inGroup($group)
+    {
+        if (is_numeric($group)) {
+            $groupCheckObj = Group::get()->byID($group);
+        } elseif (is_string($group)) {
+            $SQL_group = Convert::raw2sql($group);
+            $groupCheckObj = Group::get()->filter(['Code' => $SQL_group])->first();
+        } elseif ($group instanceof Group) {
+            $groupCheckObj = $group;
+        } else {
+            throw new \InvalidArgumentException('Group::inGroup(): Wrong format for $group parameter');
+        }
+
+        if (!$groupCheckObj) {
+            return false;
+        }
+
+        return in_array($groupCheckObj->ID, $this->collateAncestorIDs());
+    }
+
+    /**
+     * Check if the group is a child of the given groups or any parent groups
+     *
+     * @param array $groups
+     * @param bool $strict set to TRUE if must be in ALL groups, or FALSE if must be in ANY
+     * @return bool Returns TRUE if the Group is a child of any of the given groups, otherwise FALSE
+     */
+    public function inGroups($groups, $strict = false)
+    {
+        foreach ($groups as $group) {
+            if ($this->inGroup($group)) {
```

Comment:
Have reworked this, I _think_ I understood what you meant? Let me know if I've gone down the wrong rabbithole 🐰